File: Barath/src/05_Longest-Common-Prefix/longestCommonPrefix.test.ts
import longestCommonPrefix from './longestCommonPrefix';

test('Test with an array of strings with a common prefix', () => {
    expect(longestCommonPrefix(["flower","flow","flight"])).toBe("fl");
});

test('Test with an array of strings with no common prefix', () => {
    expect(longestCommonPrefix(["dog","racecar","car"])).toBe("");
});

test('Test with an array containing an empty string', () => {
    expect(longestCommonPrefix(["","b"])).toBe("");
});

test('Test with an array containing only one string', () => {
    expect(longestCommonPrefix(["flower"])).toBe("flower");
});

File: Bipin/014_longest_common_prefix/longest_common_prefix_test.go
package main

import (
	"testing"
)

func TestLongestCommonPrefix(t *testing.T) {
	// Test 1: Array with a common prefix
	result := LongestCommonPrefix([]string{"flower", "flow", "flight"})
	expected := "fl"
	if result != expected {
		t.Errorf("Test 1 failed. Expected %s, got %s", expected, result)
	}

	// Test 2: Array with no common prefix
	result = LongestCommonPrefix([]string{"dog", "racecar", "car"})
	expected = ""
	if result != expected {
		t.Errorf("Test 2 failed. Expected %s, got %s", expected, result)
	}

	// Test 3: Array containing an empty string
	result = LongestCommonPrefix([]string{"", "flow", "flight"})
	expected = ""
	if result != expected {
		t.Errorf("Test 3 failed. Expected %s, got %s", expected, result)
	}

	// Test 4: Array containing only one string
	result = LongestCommonPrefix([]string{"single"})
	expected = "single"
	if result != expected {
		t.Errorf("Test 4 failed. Expected %s, got %s", expected, result)
	}
}
